client/blocks/upe/index.js

```diff
@@ -25,6 +25,7 @@ const api = new WCStripeAPI(
 const upeMethods = getPaymentMethodsConstants();
 Object.entries( getBlocksConfiguration()?.paymentMethodsConfig )
 	.filter( ( [ upeName ] ) => upeName !== 'link' )
+	.filter( ( [ upeName ] ) => upeName !== 'giropay' ) // Skip giropay as it was deprecated by Jun, 30th 2024.
```

Comment:
Looks like we're filtering out giropay in several places right before we display it. How about removing it earlier, closer to the source we use everywhere else?

Like when retrieving the [available payment methods](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L395) that are then passed to JS...

Or earlier when retrieving the [enabled payment methods](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L495C18-L495C52).

Or removing its options from the database directly on [upgrade](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/woocommerce-gateway-stripe.php#L308-L338).

Not requesting changes for this, just curious about the approach. Maybe I'm missing a scenario that's better handled this way.

Reply:
I just felt safe removing it that way because we need to keep the method for refunds. But I think your suggestions make sense. I will merge this as is and open another PR to improve it for the next release if that sounds ok!

Reply:
Oh, thanks for checking! I will work on this separately on the next days